### ISSUE TITLE
Attempting to preserve the original realtime of the subtitles.

### ISSUE THREAD

**Issue by Jamaika1** [gpac/gpac#555]
Currently should be divided recording into a twice. It is imperfect implementation or something I don't know.
_mp4box.exe -new -info -fps 29.970 -add video_1.h265 -add audio_1.aac -add subtitle_1.ttxt test2a.mp4
mp4box.exe -new -info -fps 29.970 -add video_2.h265 -add audio_2.aac -add subtitle_2.ttxt test2b.mp4
mp4box.exe -new -info -fps 29.970 -mpeg4 -force-cat -cat test2a.mp4 -cat test2b.mp4 test2.mp4_
What errors occur in adding up the realtimes? v0.6.2-rev595
"properly Times"

```
<TextSample sampleTime="00:00:00.000" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:00:05.000" xml:space="preserve">________________
________________<Style fromChar="0" toChar="16" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
<Style fromChar="17" toChar="33" styles="Italic " fontID="1" fontSize="80" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:00:15.000" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:02:38.330" xml:space="preserve">____________<Style fromChar="0" toChar="12" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:02:48.330" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:03:55.430" xml:space="preserve">_________
_________<Style fromChar="0" toChar="9" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
<Style fromChar="10" toChar="19" styles="Italic " fontID="1" fontSize="80" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:04:05.430" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:05:06.770" xml:space="preserve">____________
________________<Style fromChar="0" toChar="12" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
<Style fromChar="13" toChar="29" styles="Italic " fontID="1" fontSize="80" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:05:16.770" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:06:30.770" xml:space="preserve">_______<Style fromChar="0" toChar="7" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:06:40.770" text="" />
```

"merge times"

```
<TextSample sampleTime="00:00:00.000" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:00:05.000" xml:space="preserve">________________
________________<Style fromChar="0" toChar="16" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
<Style fromChar="17" toChar="33" styles="Italic " fontID="1" fontSize="80" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:00:15.000" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:02:38.330" xml:space="preserve">____________<Style fromChar="0" toChar="12" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:**03:50.442**" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:03:55.442" xml:space="preserve">_________
_________<Style fromChar="0" toChar="9" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
<Style fromChar="10" toChar="19" styles="Italic " fontID="1" fontSize="80" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:04:05.442" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:05:06.747" xml:space="preserve">____________
________________<Style fromChar="0" toChar="12" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
<Style fromChar="13" toChar="29" styles="Italic " fontID="1" fontSize="80" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:05:16.747" xml:space="preserve"></TextSample>
<TextSample sampleTime="00:06:30.762" xml:space="preserve">_______<Style fromChar="0" toChar="7" styles="Normal" fontID="1" fontSize="120" color="f3 e6 15 ff"/>
</TextSample>
<TextSample sampleTime="00:06:**30**.971" text="" />
```

Last record times merge containers MP4 are from outer space. Can not also use the function 'unalign-cat'.

**Comment by rbouqueau** [comment]
Hi, thanks for reporting. Would it be possible to share some sample so that we reproduce?
We have an anonymous dropbox if it can help:  https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

**Comment by Jamaika1** [comment]
I return to the subject.
I recently discovered that a joined subtitles as SRT real time agree. I understand that GPAC resigned from the subtitle TTXT. I just want to add some attention on the SRT.
Subtitles can not have empty lines, I do not know why.
_1
00:02:38,330 --> 00:02:48,330
<-- I can't use 'Enter'
<i>text</i>_
`Bad SRT formatting - expecting number got "<i>text</i>"`
When I join different tracks group it might not be a break in the subtitle track, for ex.
_-catx "image1.h265 +image1.aac +subtitle1.srt +chapter1.txt":noedit:trailing:xps_inband:name="" -catx "image2.h265 +image2.aac +chapter2.txt" -catx "image3.h265 +image3.aac +subtitle3.srt +chapter3.txt"_
At the end of a trifle. I have a problem with the removal of descriptions tracks [:name=""].

**Comment by rbouqueau** [comment]
Thanks for reporting.

> I understand that GPAC resigned from the subtitle TTXT

I am not sure to understand your comment. We are always happy to see issues reported.

> Subtitles can't have empty lines, I don't know why.

There is no spec for SRT but it seems not legal: http://forum.doom9.org/showthread.php?t=73953

> I have a problem with the removal of descriptions tracks [-catx "image1.h265 +image1.aac +subtitle1.srt +chapter1.txt":noedit:trailing:xps_inband:name=""].

What do you call "description tracks"? How can we help? Could you please provide some samples so that we can reproduce?

**Comment by Jamaika1** [comment]
> I am not sure to understand your comment. We are always happy to see issues reported.

This works fine. Real times subtitles SRT are good.
mp4box.exe -new -info -fps 29.970p 
-catx "image1.h265 +image1.aac **+subtitle1.srt** +chapter1.txt":noedit:trailing:xps_inband:name="" 
-catx "image2.h265 +image2.aac **+subtitle2.srt** +chapter2.txt" 
-catx "image3.h265 +image3.aac **+subtitle3.srt** +chapter3.txt" 
-catx "image4.h265 +image4.aac **+subtitle4.srt** +chapter4.txt" 
-catx "image5.h265 +image5.aac" 
-catx "image6.h265 +image6.aac **+subtitle6.srt** +chapter6.txt" image.mp4
It doesn't work properly. Real times subtitles TTXT are bad.
mp4box.exe -new -info -fps 29.970p 
-catx "image1.h265 +image1.aac **+subtitle1.ttxt** +chapter1.txt":noedit:trailing:xps_inband:name="" 
-catx "image2.h265 +image2.aac **+subtitle2.ttxt** +chapter2.txt" 
-catx "image3.h265 +image3.aac **+subtitle3.ttxt** +chapter3.txt" 
-catx "image4.h265 +image4.aac **+subtitle4.ttxt** +chapter4.txt" 
-catx "image5.h265 +image5.aac" 
-catx "image6.h265 +image6.aac **+subtitle6.ttxt** +chapter6.txt" image.mp4

**Comment by rbouqueau** [comment]
> OK, but matroska and ffmpeg can use 'enter'.
Would 'enter' be inserted in the text or ignored?

> I mean [: name =""]
It resets the handler here. What do you expect, and what do you practically get?

**Comment by Jamaika1** [comment]
> Would 'enter' be inserted in the text or ignored?

Yes, 'enter' would be inserted in the text SRT.
> It resets the handler here. What do you expect, and what do you practically get?

Earlier in the Media Info 'Title' wasn't displayed. Currently it has added space.
What are the difficulties for 'catx'. When we use the command:
-catx "video.h265 +audio.aac +...":name="" isn't implemented command for all tracks. Must be:
-catx "video.h265:name= +audio.aac:name= +...:name="

**Comment by jeanlf** [comment]
I believe the timing in cat has been fixed see #599
The handler is the nirmal behaviour, it has to be set per add/cat command.
For empty lines I just patched the code, it should now be supported

**Comment by Jamaika1** [comment]
Thanks
